austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,95 @@
+#define MINEDRONE_COLLECT 1
+#define MINEDRONE_ATTACK 2
+
+/mob/living/simple_animal/hostile/mining_drone
+	var/beacons = 15
+	var/default_hatmask
+	var/obj/item/t_scanner/adv_mining_scanner/lesser/scanner //there is no code to turn this scanner off.
+
+/mob/living/simple_animal/hostile/mining_drone/Initialize()
+	. = ..()
+	weather_immunities += "ash" // no damage from ash storms
+	stored_gun.overheat_time = 10
+
+	scanner = new(src)
+	scanner.toggle_on()
+	var/datum/action/innate/minedrone/marker_beacon/beacon_action = new()
+	beacon_action.Grant(src)
+
+/datum/action/innate/minedrone/marker_beacon
+	name = "Drop Marker"
+	button_icon_state = "mech_zoom_off"
+
+/datum/action/innate/minedrone/marker_beacon/Activate()
+	var/mob/living/simple_animal/hostile/mining_drone/M = owner
+	if((!M.beacons) || (GLOB.total_beacons >= GLOB.max_beacons)) //check that we have marker beacons left, but also that there are not 100 in the world already
+		to_chat(M, "<span class='warning'>You can not place another beacon!</span>")
+		return
+	M.beacons--
+	var/obj/structure/marker_beacon/B = new(get_turf(owner))
+	to_chat(M, "<span class='notice'>You place a [B].</span>")
+	to_chat(M, "<span class='notice'>You have [M.beacons] beacons remaining.</span>")
+
+/obj/item/slimepotion/slime/sentience/mining
+	name = "minebot AI upgrade"
+	desc = "Can be used to grant sentience to minebots."
+	icon_state = "door_electronics"
+	icon = 'icons/obj/module.dmi'
+	sentience_type = SENTIENCE_MINEBOT
```

Comment:
you really don't need to specify every single variable. only the ones you're changing.
```suggestion
/obj/item/slimepotion/slime/sentience/mining
	desc = "Can be used to grant sentience to minebots."
```

Reply:
After some thought, I am going to remove my changes to the minebot sentience card, because it should be unreachable code - there is no natural way for a player to spawn one anymore.
I think I misunderstood what effect this code would have on the object anyway.